tests/qasm3/test_depth.py

```diff
@@ -624,3 +641,26 @@ def test_qasm3_depth_branching_for_external_gates():
     result = loads(qasm3_string)
     result._external_gates = ["my_gate", "my_gate_two"]
     assert result.depth() == 2
+
+
+QASM3_DECOMPOSE_GATE_DEPTH_1 = """
+OPENQASM 3.0;
+qubit[2] q1;
+qreg q[3];
+creg c[3];
+crx (0.1) q[0], q[2];
+rccx q[0], q[1], q1[0];
+"""
+
+
+@pytest.mark.parametrize(
+    ["input_qasm_str", "before_decompose", "after_decompose"],
+    [
+        (QASM3_DECOMPOSE_GATE_DEPTH_1, 2, 25),
+    ],
+)
+def test_gate_depth_decomposable_gates(input_qasm_str, before_decompose, after_decompose):
```

Comment:
Can also add a test where we have custom gates (which should be unrolled), external gate (which should be added as depth = 1) and then this new parameter in `depth`